lib/logger/lib/logger.ex

```diff
@@ -812,20 +813,22 @@ defmodule Logger do
     end
   end
 
+  defguardp is_msg(msg) when is_binary(msg) or is_list(msg) or is_map(msg)
+
   @doc false
   def __do_log__(level, fun, metadata) when is_function(fun, 0) and is_map(metadata) do
     case fun.() do
       {msg, meta} ->
-        :logger.macro_log(%{}, level, msg, Enum.into(meta, add_elixir_domain(metadata)))
+        __do_log__(level, msg, Enum.into(meta, metadata))
 
-      msg when is_binary(msg) or is_list(msg) ->
-        :logger.macro_log(%{}, level, msg, add_elixir_domain(metadata))
+      msg ->
+        __do_log__(level, msg, metadata)
     end
   end
 
-  def __do_log__(level, chardata, metadata)
-      when (is_binary(chardata) or is_list(chardata)) and is_map(metadata) do
-    :logger.macro_log(%{}, level, chardata, add_elixir_domain(metadata))
+  def __do_log__(level, msg, metadata)
+      when is_msg(msg) and is_map(metadata) do
+    :logger.macro_log(%{}, level, msg, add_elixir_domain(metadata))
   end
 
   # # TODO: Remove that in Elixir 2.0
```

Comment:
Nitpick: I think the warning two lines below should be updated to reflect the new supported types 🙂

Reply:
Done